Swiftz/DictionaryExt.swift

```diff
@@ -6,14 +6,24 @@
 //  Copyright (c) 2014 Maxwell Swadling. All rights reserved.
 //
 
+extension Array {
+	/// Creates a dictionary from the array given a transformer function that returns 
+	/// a (key, value) pair for the dictionary, if nil is returned then the value is
+	/// not added to the dictionary.
+	public func mapKeyValues<K, V>(f : Element -> (K, V)?) -> [K: V] {
```

Comment:
Come to think of it a more fitting name might be zipMap or groupBy. A similar function is called groupBy in Scala, but to accomplish this similarly we could just take in a function f that returns a key k for the value v corresponding to that element. Pseudo code:

zip(map(f)).toDictionary

Thoughts?

Sent from my iPhone
> On Nov 13, 2015, at 5:29 AM, Robert Widmann <notifications@github.com> wrote:
> 
> In Swiftz/DictionaryExt.swift:
> 
> > @@ -6,14 +6,24 @@
> >  //  Copyright (c) 2014 Maxwell Swadling. All rights reserved.
> >  //
> >  
> > +extension Array {
> > +	/// Creates a dictionary from the array given a transformer function that returns 
> > +	/// a (key, value) pair for the dictionary, if nil is returned then the value is
> > +	/// not added to the dictionary.
> > +	public func mapKeyValues<K, V>(f : Element -> (K, V)?) -> [K: V] {
> Is there a more vivid name for this combinator? mapAssoc or mapAssociate, etc.?
> 
> —
> Reply to this email directly or view it on GitHub.
>